app/src/Page/Bcc/Edit/BusinessDecision.elm

```diff
@@ -0,0 +1,158 @@
+import Html exposing (Html, div, text)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onClick, onSubmit)
+
+import Bootstrap.Button as Button
+import Bootstrap.Card as Card
+import Bootstrap.Card.Block as Block
+import Bootstrap.Text as Text
+import Bootstrap.Form as Form
+import Bootstrap.Form.Input as Input
+import Bootstrap.Form.Textarea as Textarea
+import BoundedContext.BusinessDecision exposing (BusinessDecision(..))
+import BoundedContext.BusinessDecision exposing (Problem(..))
+
+type Msg
+    = Show 
+    | StartAddNew
+    | AddNew BusinessDecision
+    | ChangeDecisionName String
+    | ChangeDecisionDescription String
+    | CancelAdding
+    | Delete String 
+
+type ChangingModel
+  = AddingNewBusinessDecision String String (Result BoundedContext.BusinessDecision.Problem BusinessDecision)
+
+type alias Model =
+  { decisions : BusinessDecisions
+  , changingModel : Maybe ChangingModel
+  }
+
+init : BusinessDecisions -> Model
+init decisions =
+  { decisions = decisions
+  , changingModel = Nothing
+  }
+
+update : Msg -> Model -> Model
+update msg model =
+    case (msg, model.changingModel) of
+        (StartAddNew, _) ->
+            { model | changingModel = Just <| AddingNewBusinessDecision "" "" (defineBusinessDecision model.decisions "" "") }
+
+        (ChangeDecisionName name, Just (AddingNewBusinessDecision _ description _)) ->
+            { model | changingModel = Just <| AddingNewBusinessDecision name description (defineBusinessDecision model.decisions name description) }
+
+        (ChangeDecisionDescription description, Just (AddingNewBusinessDecision name _ _)) ->
+            { model | changingModel = Just <| AddingNewBusinessDecision name description (defineBusinessDecision model.decisions name description) }
+
+        (CancelAdding, _) ->
+            { model | changingModel = Nothing}
+
+        (AddNew decision, Just (AddingNewBusinessDecision _ _ _)) ->
+            let
+                newDecision = addBusinessDecision model.decisions decision
+
+            in
+                case newDecision of
+                    Ok decisions ->
+                        { model
+                        | decisions = decisions
+                        , changingModel = Nothing
+                        }
+                    Err _ -> 
+
+                        model
+
+        (Delete name, Nothing) ->
+            { model | decisions = deleteBusinessDecision model.decisions name }
+        
+        _ -> model
+
+view : Model -> Html Msg
+view model =
+    Html.div []
+    [   viewAddDecision model.changingModel |> Card.view
+    ,   Html.dl []
+        (
+            List.map viewDecision model.decisions |> List.concat
+        )
+    ]
+
+viewDecision : BusinessDecision -> List (Html Msg)
+viewDecision decision =
+  [ Html.dt []
+    [ text (getName decision)
+    , Button.button
+      [ Button.secondary
+      , Button.small
+      , Button.onClick (Delete (decision |> getName))
```

Comment:
can you add a `getId` to `BusinessDecision` so that we don't need to use 'name' for these things? (Implementation wise I think using the name as id is fine ATM, but I think referencing Decisions via an ID makes sense)

Reply:
added